src/frontend/src/optimizer/plan_node/stream_sink.rs

```diff
@@ -133,9 +236,42 @@ impl StreamSink {
         Ok(Self::new(input, sink))
     }
 
+    fn derive_iceberg_sink_distribution(
```

Comment:
I find it inconvenient to test it for Iceberg because I need to connect the catalog and get related info. 
We test the sink using `handle_explain`, which means that there isn't a place to insert the mock partition info in plan test.
- One way to test it is to mock the partition info in `blackhole` connector. But this will add the code for test in frontend, I'm not sure whether it's a good way.
```
pub async fn get_partition_compute_info(
    with_options: &WithOptions,
) -> Result<Option<PartitionComputeInfo>> {
   match connector.as_str() {
        ICEBERG_SINK => {
            ..
        }
       BLACK_HOLE => { ..//generate mock info }
        _ => Ok(None),
    }
}
```
- Another way is to modify the test way for sink, maybe we can use create_sink instead of test using explain and add a param to pass the partition info so that we can mock the partition info outside. 
https://github.com/risingwavelabs/risingwave/blob/ee08ac5cbd9e2f1042ea8b61a7c9958d81840823/src/frontend/planner_test/src/lib.rs#L550

Reply:
> I find it inconvenient to test it for Iceberg because I need to connect the catalog and get related info.

How about creating a mocked iceberg catalog?